File: rust/src/api/batch_jobs.rs
// this file is @generated
#[allow(unused_imports)]
use crate::{error::Result, models::*, Configuration};

#[derive(Default)]
pub struct BatchJobsListBatchJobsOptions {
    pub job_type: Option<BatchJobType>,

    pub status: Option<Vec<BatchJobStatus>>,

    /// Page number (0-indexed)
    pub page: Option<i32>,

    /// Number of items per page
    pub per_page: Option<i32>,
}

#[derive(Default)]
pub struct BatchJobsListBatchJobFailuresOptions {
    pub chunk_id: Option<BatchJobChunkId>,

    pub limit: Option<i32>,

    pub offset: Option<i32>,
}

pub struct BatchJobs<'a> {
    cfg: &'a Configuration,
}

impl<'a> BatchJobs<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// List batch jobs with optional filtering by type and status.
    pub async fn list_batch_jobs(
        &self,
        options: Option<BatchJobsListBatchJobsOptions>,
    ) -> Result<crate::models::BatchJobListResponse> {
        let BatchJobsListBatchJobsOptions {
            job_type,
            status,
            page,
            per_page,
        } = options.unwrap_or_default();

        crate::request::Request::new(http1::Method::GET, "/api/v1/batch-jobs")
            .with_optional_query_param("job_type", job_type)
            .with_optional_query_param("status", status)
            .with_optional_query_param("page", page)
            .with_optional_query_param("per_page", per_page)
            .execute(self.cfg)
            .await
    }

    /// Retrieve a single batch job with its chunks and failures.
    pub async fn get_batch_job(
        &self,
        batch_job_id: String,
    ) -> Result<crate::models::BatchJobDetailResponse> {
        crate::request::Request::new(http1::Method::GET, "/api/v1/batch-jobs/{batch_job_id}")
            .with_path_param("batch_job_id", batch_job_id)
            .execute(self.cfg)
            .await
    }

    /// Retrieve paginated failures for a batch job.
    pub async fn list_batch_job_failures(
        &self,
        batch_job_id: String,
        options: Option<BatchJobsListBatchJobFailuresOptions>,
    ) -> Result<crate::models::BatchJobFailuresResponse> {
        let BatchJobsListBatchJobFailuresOptions {
            chunk_id,
            limit,
            offset,
        } = options.unwrap_or_default();

        crate::request::Request::new(
            http1::Method::GET,
            "/api/v1/batch-jobs/{batch_job_id}/failures",
        )
        .with_path_param("batch_job_id", batch_job_id)
        .with_optional_query_param("chunk_id", chunk_id)
        .with_optional_query_param("limit", limit)
        .with_optional_query_param("offset", offset)
        .execute(self.cfg)
        .await
    }
}

File: typescript/src/api/batchJobs.ts
// this file is @generated
import type { BatchJobChunkId } from "../models/batchJobChunkId";
import {
  type BatchJobDetailResponse,
  BatchJobDetailResponseSerializer,
} from "../models/batchJobDetailResponse";
import {
  type BatchJobFailuresResponse,
  BatchJobFailuresResponseSerializer,
} from "../models/batchJobFailuresResponse";
import {
  type BatchJobListResponse,
  BatchJobListResponseSerializer,
} from "../models/batchJobListResponse";
import type { BatchJobStatus } from "../models/batchJobStatus";
import type { BatchJobType } from "../models/batchJobType";
import { HttpMethod, MeteroidRequest, type MeteroidRequestContext } from "../request";

export interface BatchJobsListBatchJobsOptions {
  jobType?: BatchJobType;
  status?: BatchJobStatus[];
  /** Page number (0-indexed) */
  page?: number;
  /** Number of items per page */
  perPage?: number;
}

export interface BatchJobsListBatchJobFailuresOptions {
  chunkId?: BatchJobChunkId;
  limit?: number;
  offset?: number;
}

export class BatchJobs {
  public constructor(private readonly requestCtx: MeteroidRequestContext) {}

  /** List batch jobs with optional filtering by type and status. */
  public listBatchJobs(
    options?: BatchJobsListBatchJobsOptions
  ): Promise<BatchJobListResponse> {
    const request = new MeteroidRequest(HttpMethod.GET, "/api/v1/batch-jobs");

    request.setQueryParam("job_type", options?.jobType);
    request.setExplodedQueryParam("status", options?.status);
    request.setQueryParam("page", options?.page);
    request.setQueryParam("per_page", options?.perPage);
    return request.send(this.requestCtx, BatchJobListResponseSerializer._fromJsonObject);
  }

  /** Retrieve a single batch job with its chunks and failures. */
  public getBatchJob(batchJobId: string): Promise<BatchJobDetailResponse> {
    const request = new MeteroidRequest(
      HttpMethod.GET,
      "/api/v1/batch-jobs/{batch_job_id}"
    );

    request.setPathParam("batch_job_id", batchJobId);
    return request.send(
      this.requestCtx,
      BatchJobDetailResponseSerializer._fromJsonObject
    );
  }

  /** Retrieve paginated failures for a batch job. */
  public listBatchJobFailures(
    batchJobId: string,
    options?: BatchJobsListBatchJobFailuresOptions
  ): Promise<BatchJobFailuresResponse> {
    const request = new MeteroidRequest(
      HttpMethod.GET,
      "/api/v1/batch-jobs/{batch_job_id}/failures"
    );

    request.setPathParam("batch_job_id", batchJobId);
    request.setQueryParam("chunk_id", options?.chunkId);
    request.setQueryParam("limit", options?.limit);
    request.setQueryParam("offset", options?.offset);
    return request.send(
      this.requestCtx,
      BatchJobFailuresResponseSerializer._fromJsonObject
    );
  }
}
